### PR TITLE
[FIX] mail: fix tracking value issue when value is 0

### PR DESCRIPTION
**Pull request by pka-odoo**
Before this commit:

When a tracking field is set to 0, the mail tracking value doesn't display 0 in
the chatter. It is empty instead
i.e:  Field Name: 1 ->

After this commit:

Tracking value will show the changed value even if it is false or 0
i.e: Field name: 1 -> 0

Task:
https://www.odoo.com/web#action=4043&cids=2&id=2334048&menu_id=4720&model=project.task&view_type=form

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by seb-odoo** [comment]
Hello, thanks for the PR.

Please add tests for this, at least covering:
- false (boolean)
- 0 (integer)
- "" (empty string)

**Comment by pka-odoo** [comment]
@seb-odoo testcase is done. Can you please check? 
thanks

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 1]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
 });
 
+QUnit.test('message list for tracking field', async function (assert) {
+    assert.expect(3);
+    await this.start();
+    this.data["mail.message"].fields.tracking_value_ids = {
+        relation: "mail.tracking.values",
+        string: "Tracking Values",
+        type: "many2many",
+    }
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:441]: Add missing field in `mock_models` file.

This is the result of a formatter though, so it's not a related field in JS. It's just an array of objects with a specific shape.

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 2]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
+        model: 'mail.channel',
+        res_id: 100,
+        id: 1,
+        tracking_value_ids: [{
+            changed_field: "Field 1",
+            field_type: "float",
+            id: 6,
+            new_value: 0,
+            old_value: 1,
+        }],
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:468]: Create thread and message in `this.data` before start (there are other examples in the code). Other tests in this file have to be adapted in another PR.

The goal is that if they are defined before start, they will be received at init messaging. Then the thread model can be found with `findFromIdentifyingData` instead of `create`

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 3]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
+    assert.strictEqual(
+        document.querySelectorAll('.o_MessageList_message .o_Message_content')[1].textContent,
+        "Field 1:truefalse",
+        "Message content for boolean type field should be 'Field 1:truefalse'"
+    );
+    assert.strictEqual(
+        document.querySelectorAll('.o_MessageList_message .o_Message_content')[2].textContent,
+        "Field 1:Name",
+        "Message content for char type field should be 'Field 1:Name'"
+    );
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:514]: Make 3 different simpler tests, one for each case.

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 4]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
+
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView);
+    assert.strictEqual(
+        document.querySelectorAll('.o_MessageList_message .o_Message_content')[0].textContent,
+        "Field 1:10",
+        "Message content for float type field should be 'Field 1:10"
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:503]: this notation is really confusing without space and separator icon, in the assert comment maybe make it more visual such as:  'Field 1: 1 -> 0'
(same for others)

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 5]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
+        public: 'public',
+    });
+
+    this.data['mail.message'].records.push({
+        channel_ids: [100],
+        model: 'mail.channel',
+        res_id: 100,
+        id: 1,
+        tracking_value_ids: [{
+            changed_field: "Field 1",
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:463]: Rename to just "Field" (if you split in 3 tests you just need 1 field per test).

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 6]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
+        model: 'mail.channel',
+        name: "General",
+        public: 'public',
+    });
+
+    this.data['mail.message'].records.push({
+        channel_ids: [100],
+        model: 'mail.channel',
+        res_id: 100,
+        id: 1,
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:461]: don't need to specify id if you don't use it.

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 7]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
+        string: "Tracking Values",
+        type: "many2many",
+    }
+    const thread = this.env.models['mail.thread'].create({
+        channel_type: 'channel',
+        id: 100,
+        members: [['insert', [
+            {
+                email: "john@example.com",
+                id: 9,
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:448]: prefer ID value above 10 to avoid conflicts with potential "default" data from server (this is purely arbitrary)

note that in this case you can remove this. Just put thread in this.data before start, and current partner will be member by default which will be enough for the test.

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 8]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
+        document.querySelectorAll('.o_MessageList_message .o_Message_content')[1].textContent,
+        "Field 1:truefalse",
+        "Message content for boolean type field should be 'Field 1:truefalse'"
+    );
+    assert.strictEqual(
+        document.querySelectorAll('.o_MessageList_message .o_Message_content')[2].textContent,
+        "Field 1:Name",
+        "Message content for char type field should be 'Field 1:Name'"
+    );
+
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:515]: no new line at the end here

File: addons/mail/static/src/components/thread_view/thread_view_tests.js [hunk 9]
```diff
@@ -431,6 +431,91 @@ QUnit.test('show message subject if thread is mailing channel', async function (
...
+});
+
+
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view_tests.js:518]: and here only 1 new line not 2

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Good progress, here's a quick check to request some reorganization in test.

**Comment by pka-odoo** [comment]
Hello @seb-odoo Changes has been done. Thanks

**Comment by seb-odoo** [comment]
Thank you, I'll check and merge it today.

File: addons/mail/static/src/components/message/message_tests.js [hunk 10]
```diff
@@ -984,6 +984,195 @@ QUnit.test('open chat with author on avatar click should be disabled when curren
...
+        "should display the new value"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_Message_trackingValueNewValue').textContent,
+        "45.67",
+        "should display the correct new value (45.67)",
+    );
+});
+
+QUnit.test('rendering of tracking value (float type) when changing from non-0 to 0', async function (assert) {
```
Comment (alexkuhn) [inline addons/mail/static/src/components/message/message_tests.js:1044]: _"rendering of tracked field with change of value from non-0 to 0"_

Comment (seb-odoo) [reply addons/mail/static/src/components/message/message_tests.js]: done

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
